.github/workflows/test_e2e.yml

```
@@ -102,3 +103,123 @@ jobs:
         with:
           name: debug-logs-${{ env.SCOPE }}
           path: debug-logs/
+
+  robot:
+    name: Robot
+    permissions:
+      id-token: write
+
+    # Make sure that only one Job is using the server at a time
+    concurrency: robot-test-server
+    environment: e2e-robot
+
+    env:
+      K3S_CHANNEL: v1.28
+      SCOPE: gha-${{ github.run_id }}-${{ github.run_attempt }}-robot
+
+      # Disable routes in dev-env, not supported for Robot.
+      ROUTES_ENABLED: "false"
+      ROBOT_ENABLED: "true"
+      SERVER_NUMBER: ${{ vars.SERVER_NUMBER }}
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
+      - uses: actions/checkout@master
+      - uses: hetznercloud/tps-action@main
+        with:
+          token: ${{ secrets.HCLOUD_TOKEN  }}
+      - uses: 3bit/setup-hcloud@v2
+      - uses: yokawasa/action-setup-kube-tools@v0.9.3
+        with:
+          setup-tools: |
+            helm
+            kubectl
+            skaffold
+          helm: v3.11.2
+          kubectl: v1.28.1
+          skaffold: v2.3.0
+      - name: Install k3sup
+        run: |
+          curl -sLS https://get.k3sup.dev | sh
+
+      - name: Setup test environment
+        env:
+          ROBOT_USER: ${{ secrets.ROBOT_USER }}
+          ROBOT_PASSWORD: ${{ secrets.ROBOT_PASSWORD }}
+        run: |
+          source <(hack/dev-up.sh)
+          
+          # make exported env variables available to following jobs
+          echo "KUBECONFIG=$KUBECONFIG" >> "$GITHUB_ENV"
+          echo "SKAFFOLD_DEFAULT_REPO=$SKAFFOLD_DEFAULT_REPO" >> "$GITHUB_ENV"
+          echo "CONTROL_IP=$CONTROL_IP" >> "$GITHUB_ENV"
+
+      - name: Build and Deploy HCCM
+        run: |
+          skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}"
+          tag=$(skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}" --quiet --output="{{ (index .Builds 0).Tag }}")
+          skaffold deploy \
+            --profile=robot \
+            --images=hetznercloud/hcloud-cloud-controller-manager=$tag
+
+      - name: Setup Robot Server
+        env:
+          ROBOT_USER: ${{ secrets.ROBOT_USER }}
+          ROBOT_PASSWORD: ${{ secrets.ROBOT_PASSWORD }}
+        working-directory: hack/robot-e2e
+        run: |
+          ansible-galaxy install -r requirements.yml
+          echo "::group::ansible-playbook e2e-setup-robot-server.yml"
+          ansible-playbook e2e-setup-robot-server.yml -e scope=$SCOPE -e server_number=$SERVER_NUMBER -vvv
+          echo "::endgroup::"
```

```suggestion
      - name: Setup Robot Server
        env:
          ROBOT_USER: ${{ secrets.ROBOT_USER }}
          ROBOT_PASSWORD: ${{ secrets.ROBOT_PASSWORD }}
          PY_COLORS: true
          ANSIBLE_FORCE_COLOR: true
        working-directory: hack/robot-e2e
        run: |
          ansible-galaxy install -r requirements.yml
          echo "::group::ansible-playbook e2e-setup-robot-server.yml"
          ansible-playbook e2e-setup-robot-server.yml -e scope=$SCOPE -e server_number=$SERVER_NUMBER -vvv
          echo "::endgroup::"
```

This should add some colors the ansible logs, and help to debug.